tests/Dialect/Polynomial/Conversions/heir_polynomial_to_llvm/lower_eval.mlir

```diff
@@ -0,0 +1,13 @@
+// RUN: heir-opt --polynomial-to-mod-arith %s | FileCheck %s
+
+!eval_poly_ty = !polynomial.polynomial<ring=<coefficientType=i64>>
+#eval_poly = #polynomial.typed_int_polynomial<4+x+x**2> : !eval_poly_ty
+
+func.func @test_eval() -> i64 {
+    %c6 = arith.constant 6 : i64
+    %0 = polynomial.eval #eval_poly, %c6 : i64
```

Comment:
Please add `// CHECK` directives here for the expected output ops. See https://github.com/google/heir/blob/c90c3e26fcb909a0620261157f21bb4f34f7741e/tests/Dialect/Polynomial/IR/canonicalization.mlir as a good example of the FileCheck syntax using regular expressions to capture SSA variable names.